examples/run_CPMG.py

```diff
@@ -1,95 +1,117 @@
+    nPhaseSteps = 1
+# }}}
+# {{{run cpmg
 # NOTE: Number of segments is nEchoes * nPhaseSteps
 data = run_cpmg(
-        nScans=nScans,
-        indirect_idx = 0,
-        ph1_cyc = ph1_cyc,
-        adcOffset = adcOffset,
-        carrierFreq_MHz = carrierFreq_MHz,
-        nPoints = nPoints,
-        nEchoes = nEchoes,
-        p90_us = p90_us,
-        repetition = repetition_us,
-        tau_us = tau_us,
-        SW_kHz = SW_kHz,
-        pad_start_us = pad_start,
-        pad_end_us = pad_end,
-        output_name = output_name,
-        ret_data = None)
-acq_params = {j: eval(j) for j in dir() if j in [
-        "adcOffset",
-        "carrierFreq_MHz",
-        "amplitude",
-        "nScans",
-        "nEchoes",
-        "p90_us",
-        "deadtime_us",
-        "repetition_us",
-        "SW_kHz",
-        "nPoints",
-        "deblank_us",
-        "tau_us",
-        "nPhaseSteps",
-        ]
-        }
-data.set_prop("acq_params",acq_params)
-data.name(node_name)
-if phase_cycling:
-    data.chunk("t",['ph1','t2'],[len(ph1_cyc),-1])
-    data.setaxis('ph1', ph1_cyc/4)
-data.hdf5_write(date + '_'+output_name+'.h5',
-        directory=getDATADIR(exp_type = 'ODNP_NMR_comp/CPMG'))
+    nScans=config_dict["nScans"],
+    indirect_idx=0,
+    ph1_cyc=ph1_cyc,
+    adcOffset=config_dict["adc_offset"],
+    carrierFreq_MHz=config_dict["carrierFreq_MHz"],
+    nPoints=nPoints,
+    nEchoes=config_dict["nEchoes"],
+    p90_us=config_dict["p90_us"],
+    repetition=config_dict["repetition_us"],
+    tau_us=config_dict["tau_us"],
+    SW_kHz=config_dict["SW_kHz"],
+    pad_start_us=pad_start,
+    pad_end_us=pad_end,
+    output_name=filename,
+    ret_data=None,
+)
+# }}}
+# {{{saving with acq params
 SpinCore_pp.stopBoard();
+data.set_prop("acq_params", config_dict.asdict())
+data.name(config_dict["type"] + "_" + config_dict["cpmg_counter"])
+target_directory = getDATADIR(exp_type="ODNP_NMR_comp/CPMG")
+filename_out = filename + ".h5"
+nodename = data.name()
+if phase_cycling:
+    data.chunk("t", ["ph1", "t2"], [len(ph1_cyc), -1])
+    data.setaxis("ph1", ph1_cyc / 4)
+    if config_dict["nScans"] > 1:
+        data.setaxis("nScans", r_[0 : config_dict["nScans"]])
+if os.path.exists(filename + ".h5"):
+    print("this file already exists so we will add a node to it!")
+    with h5py.File(
+        os.path.normpath(os.path.join(target_directory, f"{filename_out}"))
+    ) as fp:
+        if nodename in fp.keys():
+            print("this nodename already exists, so I will call it temp")
+            data.name("temp")
+            nodename = "temp"
+    data.hdf5_write(f"{filename_out}", directory=target_directory)
+else:
+    try:
+        data.hdf5_write(f"{filename_out}", directory=target_directory)
+    except:
+        print(
+            f"I had problems writing to the correct file {filename}.h5, so I'm going to try to save your file to temp.h5 in the current directory"
+        )
+        if os.path.exists("temp.h5"):
```

Comment:
if the file doesn't exist and writing a new one doesn't work just change the filename